### PR TITLE
[docs-api][s]: updated api section for new DataHub - refs #51

### PR DESCRIPTION
**Pull request by Mikanebu**
* Updated API section for new DataHub
  * Introduction
  * Auth API
  * Rawstore API
  * Sources API

### CODE REVIEW

File: docs/developers/api.md [hunk 1]
```diff
@@ -1,29 +1,323 @@
...
-window.onload = function() {
-  // Build a system
-  const ui = SwaggerUIBundle({
-    url: "https://staging.datapackaged.com/api/swagger.json",
-    dom_id: '#swagger-ui',
-    validatorUrl : null,
-    presets: [
-      SwaggerUIBundle.presets.apis,
-      SwaggerUIStandalonePreset
+The DataHub API provides a range of endpoints to interact with the platform. All endpoints live under the URL `https://api.datahub.io` where our API is divided into the following sections: **auth, rawstore, sources, assemblers.**
```
Comment (rufuspollock) [inline docs/developers/api.md]: I wouldn't mention assemblers when we don't have it yet

File: docs/developers/api.md [hunk 2]
```diff
@@ -1,29 +1,323 @@
...
+            "url": "<url-for-logging-in-with-the-Github-provider>"
+        },
+    }
+}
+```
+
+When the authentication flow is finished, the caller will be redirected to the `next` URL with an extra query parameter
+`jwt` which contains the authentication token. The caller should cache this token for further interactions with the API.
+
+### Get permission for a service
```
Comment (rufuspollock) [inline docs/developers/api.md:59]: Space after heading

File: docs/developers/api.md [hunk 3]
```diff
@@ -1,29 +1,323 @@
...
+**Returns:**
+
+```json
+{
+    "success": true,
+    "error": "<error-message-if-applicable>"
+}
+```
+
+__Note__: trying to update other user profile fields like `email` will fail silently and return
```
Comment (rufuspollock) [inline docs/developers/api.md]: I would use `**Note**` rather than `__Note__`

File: docs/developers/api.md [hunk 4]
```diff
@@ -1,29 +1,323 @@
...
+
+```json
```
Comment (rufuspollock) [inline docs/developers/api.md]: Why json here vs javascript elsewhere. Use one consistently.

File: docs/developers/api.md [hunk 5]
```diff
@@ -1,29 +1,323 @@
...
+    "userid": "<unique-id-of-the-user>",
+    "permissions": {
+        "permission-x": true,
+        "permission-y": false
+    },
+    "service": "<relevant-service>"
+}
+```
+
+### Change the username
```
Comment (rufuspollock) [inline docs/developers/api.md:84]: Space after heading again.

File: docs/developers/api.md [hunk 6]
```diff
@@ -1,29 +1,323 @@
...
+{
+  "success": true,
+  "id": "<identifier>"
+  "errors": [
+      "<error-message>"
+  ]
 }
-</script>
+```
+
```
Comment (rufuspollock) [inline docs/developers/api.md]: One trailing space please :-)

**Review by rufuspollock — COMMENTED** [review]
Looks good - some very minor formatting to fix.

File: docs/developers/api.md (hunk 1 above)
Comment (Mikanebu) [reply docs/developers/api.md]: done

File: docs/developers/api.md (hunk 2 above)
Comment (Mikanebu) [reply docs/developers/api.md:59]: ok

File: docs/developers/api.md (hunk 5 above)
Comment (Mikanebu) [reply docs/developers/api.md:84]: ok

File: docs/developers/api.md (hunk 3 above)
Comment (Mikanebu) [reply docs/developers/api.md]: ok

File: docs/developers/api.md (hunk 6 above)
Comment (Mikanebu) [reply docs/developers/api.md]: ok

File: docs/developers/api.md (hunk 4 above)
Comment (Mikanebu) [reply docs/developers/api.md]: replaced all json with javascript

**Pull request merged**